src/NotificationSubscriber.ts

```diff
@@ -0,0 +1,65 @@
+
+import { Logger, LoggerFactory } from '@luckbox/logger-factory';
+import Redis, { RedisOptions } from 'ioredis';
+
+export type MessageHandler = (namespace: string, event: string, message: unknown, rooms: Array<string>, userIdOrSocketId: string | number | null) => any;
+
+type ConstructorParams = {
+  prefix: string,
+  handler: MessageHandler,
+  redisConfig: RedisOptions,
+  loggerFactory: LoggerFactory,
+};
+
+export default class NotificationSubscriber {
+  private namespaces: Record<string, string> = {};
+  private logger: Logger;
+  private redisClient: Redis;
+  private prefix: string;
+  private handler: MessageHandler;
+
+  constructor(params: ConstructorParams) {
+    this.logger = params.loggerFactory.create(this.constructor.name);
+    this.prefix = params.prefix;
+    this.handler = params.handler;
+
+    this.redisClient = new Redis(params.redisConfig);
+  }
+
+  subscribe(): void {
+    this.redisClient.on('message', (channel, packet) => {
+      if (channel === `${this.prefix}/tx`) {
+        this.register(packet);
+      } else if (channel in this.namespaces) {
+        try {
+          this.receivePacket(channel, packet);
+        } catch (error) {
+          this.logger.warn(error);
+        }
+      }
+    });
+    this.redisClient.subscribe(`${this.prefix}/tx`);
```

Comment:
It would be safer to await for these invocations as the Redis will throw if something is wrong.